src/plugins/orangehrmClaimPlugin/Api/ClaimExpenseTypeAPI.php

```diff
@@ -162,50 +243,144 @@ private function setExpenseType(ExpenseType $expenseType): void
     {
-        throw $this->getNotImplementedException();
+        $id = $this->getRequestParams()->getInt(RequestParams::PARAM_TYPE_ATTRIBUTE, CommonParams::PARAMETER_ID);
+        $expenseType = $this->getClaimService()->getClaimDao()->getExpenseTypeById($id);
+        $this->throwRecordNotFoundExceptionIfNotExist($expenseType, ExpenseType::class);
+        return new EndpointResourceResult(ClaimExpenseTypeModel::class, $expenseType);
     }
 
     /**
      * @inheritDoc
      */
     public function getValidationRuleForGetOne(): ParamRuleCollection
     {
-        throw $this->getNotImplementedException();
+        return new ParamRuleCollection(
+            new ParamRule(
+                CommonParams::PARAMETER_ID,
+                new Rule(Rules::POSITIVE)
+            ),
+        );
     }
 
     /**
+     * @OA\Put(
+     *     path="/api/v2/claim/expenses/types/{id}",
+     *     tags={"Claim/ExpenseTypes"},
+     *     @OA\PathParameter(
+     *         name="id",
+     *         @OA\Schema(type="integer")
+     *     ),
+     *     @OA\RequestBody(
+     *         required=true,
+     *         @OA\JsonContent(
+     *             @OA\Property(
+     *                 property="description",
+     *                 type="string"
+     *             ),
+     *             @OA\Property(
+     *                 property="status",
+     *                 type="boolean"
+     *             )
+     *         )
+     *     ),
+     *     @OA\Response(
+     *         response="200",
+     *         description="Success",
+     *         @OA\JsonContent(
+     *             @OA\Property(
+     *                 property="data",
+     *                 ref="#/components/schemas/Claim-ClaimExpenseTypeModel"
+     *             ),
+     *             @OA\Property(property="meta", type="object")
+     *         )
+     *     ),
+     *     @OA\Response(response="404", ref="#/components/responses/RecordNotFound")
+     * )
      * @inheritDoc
      */
     public function update(): EndpointResult
     {
-        throw $this->getNotImplementedException();
+        $id = $this->getRequestParams()->getInt(RequestParams::PARAM_TYPE_ATTRIBUTE, CommonParams::PARAMETER_ID);
+        $expenseType = $this->getClaimService()->getClaimDao()->getExpenseTypeById($id);
+        $this->throwRecordNotFoundExceptionIfNotExist($expenseType, ExpenseType::class);
+        $this->setExpenseType($expenseType);
+        $this->getClaimService()->getClaimDao()->saveExpenseType($expenseType);
+        return new EndpointResourceResult(ClaimExpenseTypeModel::class, $expenseType);
     }
 
     /**
      * @inheritDoc
      */
     public function getValidationRuleForUpdate(): ParamRuleCollection
     {
-        throw $this->getNotImplementedException();
+        return new ParamRuleCollection(
+            new ParamRule(
+                CommonParams::PARAMETER_ID,
+                new Rule(Rules::POSITIVE)
+            ),
+            new ParamRule(
```

Comment:
I feel can move description and status validations to a method, so create and update validation can reuse.
@ManulMax check this

Reply:
better if we can since its using in other places